generate-stackbrew-library.sh

```diff
@@ -9,37 +9,47 @@ array_6_6='6 latest';
 
 cd $(cd ${0%/*} && pwd -P);
 
+self="$(basename "$BASH_SOURCE")"
+
 versions=( */ )
 versions=( "${versions[@]%/}" )
-url='git://github.com/nodejs/docker-node'
+url='https://github.com/nodejs/docker-node'
+
+# sort version numbers with highest first
+IFS=$'\n'; versions=( $(echo "${versions[*]}" | sort -r) ); unset IFS
+
+# get the most recent commit which modified any of "$@"
+fileCommit() {
+	git log -1 --format='format:%H' HEAD -- "$@"
+}
 
-echo '# maintainer: Node.js Docker Team <https://github.com/nodejs/docker-node> (@nodejs)'
+echo "# this file is generated via ${url}/blob/$(fileCommit "$self")/$self"
+echo
+echo "Maintainers: The Node.js Docker Team <${url}> (@nodejs)"
+echo "GitRepo: ${url}.git"
+echo
 
 for version in "${versions[@]}"; do
 	if [[ "$version" == "docs" ]]; then
 		continue
 	fi
 	eval stub=$(echo "$version" | awk -F. '{ print "$array_" $1 "_" $2 }');
-	commit="$(git log -1 --format='format:%H' -- "$version")"
+	commit="$(fileCommit "$version")"
 	fullVersion="$(grep -m1 'ENV NODE_VERSION ' "$version/Dockerfile" | cut -d' ' -f3)"
 
 	versionAliases=( $fullVersion $version ${stub} )
-
+	echo "Tags: ${versionAliases[@]}"
+	echo "GitCommit: ${commit}"
+	echo "Directory: ${version}"
 	echo
-	for va in "${versionAliases[@]}"; do
-		echo "$va: ${url}@${commit} $version"
-	done
 
-	for variant in onbuild slim wheezy; do
-		commit="$(git log -1 --format='format:%H' -- "$version/$variant")"
+	variants=$(ls -d $version/*/ | awk -F"/" '{print $2}')
```

Comment:
Depending on your local `basename`, this could be simplified as `$(basename $version/*/)`, or if you're unfortunate enough to have a more POSIXy `basename` (or are worried about compatibility), something like `$(echo $version/*/ | xargs basename)`.

(I personally try to avoid scripting against the output of `ls` since it's usually so human-readability-focused, and Bash usually has some other more reliable method to accomplish what I'm after. :smile:)